src/circleci/rollcage/core.clj

```diff
@@ -346,27 +346,25 @@
         (report-uncaught-exception "critical" client ex thread)
         (handler thread ex))))))
 
-(def critical
-  "Notify Rollbar of an exception with level `critical`.
-  See the [[notify]] function for more information."
-  (partial notify "critical"))
-
-(def error
-  "Notify Rollbar of an exception with level `error`.
-  See the [[notify]] function for more information."
-  (partial notify "error"))
-
-(def warning
-  "Notify Rollbar of an exception with level `warning`.
-  See the [[notify]] function for more information."
-  (partial notify "warning"))
-
-(def info
-  "Notify Rollbar of an exception with level `info`.
-  See the [[notify]] function for more information."
-  (partial notify "info"))
-
-(def debug
-  "Notify Rollbar of an exception with level `debug`.
-  See the [[notify]] function for more information."
-  (partial notify "debug"))
+(defmacro deflevel
```

Comment:
I'm not a fan of the top-level `do` that that modifies the var after creation, and and duplication of the `arglists` of `notifiy`.

How about something like this:

```Clojure
(defmacro deflevel
  [level]
  `(def ~(with-meta level
           {:arglists (quote (-> #'notify meta :arglists))
            :doc (str "Notify Rollbar of an exception with level `" level "`.\n  "
                      "See the [[notify]] function for more information")})
     (partial notify ~(str level))))
```

Reply:
that's much better indeed, PR updated